From 698159a5742cdb6a29ef0947c2a52ee4b66edb0f Mon Sep 17 00:00:00 2001
From: Varsha Maira <vmaira@habana.ai>
Date: Mon, 22 Apr 2024 14:34:51 +0300
Subject: [PATCH] [SW-180536] gaudi3: tpm: move pif arc fw inter buffer

During cold boot PIF ARC firmware is copied from SPI
flash to ARC1 ICCM, and later copied to the PIF CCMs.
This overwites the manifest image that is copied to
ARC1 ICCM by bootrom.

For Gaudi3, we extend the PCR of the manifest hash
alone, instead of separately extending PCRs for
individual ppboot and preboot images. In order to
prevent the manifest image overwrite during PIF ARC
firmware copy, the buffer used to copy PIF ARC
firmware is moved from ICCM base by 64K offset.

Change-Id: I73fb6e64350b26f7d273a59f8d1824e324d6f1de
Signed-off-by: Varsha Maira <vmaira@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 7b6c57e..a76bb57 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -1151,7 +1151,7 @@
 		.cpu_cfg_base = mmD0_PIF_ARC_BASE,
 		.img_type[PIF_ARC_ICCM_IDX] = HL_FW_IMG_D0_PIF_ICCM,
 		.img_type[PIF_ARC_DCCM_IDX] = HL_FW_IMG_D0_PIF_DCCM,
-		.inter_buffer_addr = ARC1_ICCM_FULL_BASE,
+		.inter_buffer_addr = ARC1_ICCM_FULL_BASE + SZ_64K,
 		.die_id = GAUDI3_MASTER_DIE_ID,
 	};
 
